core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -460,7 +460,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |                              on one of the worker machines inside the cluster ("cluster")
         |                              (Default: client).
         |  --class CLASS_NAME          Your application's main class (for Java / Scala apps).
-        |  --name NAME                 A name of your application.
+        |  --name NAME                 A name of your application. In yarn-cluster mode the name
```

Comment:
I'm not a big fan of mentioning YARN here since this parameter if for all modes. Also, I don't think the documentation should encourage the use of `SparkContext.setAppName`, exactly because it has different behaviour in client and cluster modes.

In summary, I think this comment should be left alone. If `--name` for some reason doesn't work in yarn-client mode, that should be fixed. If there are other documents that talk about `SparkContext.setAppName`, they should probably be changed too.